include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -698,45 +719,114 @@ class graph_oriented_layout_design_impl
         {
             timeout = 10000u;
         }
-        bool timeout_limit_reached = false;
 
-        // main A* loop
+        // main loop
         while (!timeout_limit_reached)
         {
-            for (auto& ssg : ssg_vec)
+            // if multithreading is enabled
+            if (ps.enable_multithreading)
             {
-                if (ssg.frontier_flag)
+                // mutex to protect shared resources (if necessary)
+                std::mutex                                   update_best_layout_mutex{};
+                std::vector<std::future<std::optional<Lyt>>> futures;
+                futures.reserve(ssg_vec.size());
+
+                // process `ssg_vec` in parallel using std::async
+                for (auto& ssg : ssg_vec)
                 {
-                    num_evaluated_paths++;
-                    const auto expansion = expand(ssg);
-                    if (expansion.second)
+                    futures.emplace_back(
+                        std::async(std::launch::async,
+                                   [&]() -> std::optional<Lyt>  // return std::optional<layout>
```

Comment:
Code comments should explain "why" a code does something or break down complex code with natural language. This one seems to simply repeat the code in natural language. Is it necessary? Can it be replaced with a reasoning?